### PR TITLE
109 users can browse and filter through storylines

### PR DESCRIPTION
**Pull request by pascalalferink**
@antw @mattijsstam @broekhuisg  could one of you review this branch?

### CODE REVIEW

File: frontend/components/Storyline/StorylineOverview/StorylineOverview.tsx [hunk 1]
```diff
@@ -63,35 +99,30 @@ export default function StorylineOverview() {
...
-              <StorylineOverviewCard key={index} project={project} />
-              <StorylineOverviewCard key={index} project={project} />
-              <StorylineOverviewCard key={index} project={project} />
-              <StorylineOverviewCard key={index} project={project} />
-              <StorylineOverviewCard key={index} project={project} />
-            </>
-          ))}
+          {filteredProjects &&
+            filteredProjects.length &&
+            filteredProjects.map((project, index) => (
```
Comment (antw) [inline frontend/components/Storyline/StorylineOverview/StorylineOverview.tsx:123]:
```suggestion
          {filteredProjects?.map((project, index) => (
```

File: frontend/components/Storyline/StorylineOverview/StorylineOverviewCard.tsx [hunk 2]
```diff
@@ -1,57 +1,82 @@
...
-import RawHtml from "../RawHtml";
+import RawHtml from "../../RawHtml";
+import Image from "next/image";
+import Link from "next/link";
+import { useRouter } from "next/router";
 
 interface Props {
-  key: number;
+  index: number;
   project: {
```
Comment (antw) [inline frontend/components/Storyline/StorylineOverview/StorylineOverviewCard.tsx:10]: Ideally, we'd have a standard `Project` type somewhere that we can reuse.

`StorylineProps` and this `project` type have a lot in common, and in `StorylineOverview` we're treating them as the same (by passing a `StorylineProps` into `StorylineOverviewCard` as the `project` parameter.

**Review by antw — COMMENTED** [review]
I left a few comments about the React code. Mostly nitpicks that aren't essential to change

I'm curious about the decision to add [custom CSS](https://github.com/ZEnMo/Holon-webapp/pull/126/files#diff-8b8c096f5e82b537835f8b0b4703d0749a6a0cf70352524afdc140d6c304015b) rather than using Tailwind. Was TW not up to the job for the storyline cards? If we're going to write custom CSS, I think it would be preferable to use [CSS modules](https://nextjs.org/docs/basic-features/built-in-css-support#adding-component-level-css) rather than importing storyline.css globally.

**Comment by mattijsstam** [comment]
I have some small points with respect to the overal UX for the content manager and user

Content manager:
- [ ] I can't configure the icon 
- [ ]

**Comment by mattijsstam** [comment]
@pascalalferink @broekhuisg the linting of migraitons is ignored but now the tests are failing for this merge request

**Review by mattijsstam — CHANGES_REQUESTED** [review]

File: frontend/containers/StorylineOverviewPage/StorylineOverviewPage.test.tsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+describe('<StorylineOverviewPage />', () => {
+    it('Renders an empty StorylineOverviewPage', () => {
+        render(<StorylineOverviewPage />);
+    });
+
+    // it('Renders StorylineOverviewPage with data', () => {
+    //     const { container } = render(<StorylineOverviewPage {...data} />);
+    //     expect(container).toMatchSnapshot();
+    // });
```
[the remark this answers is not in the document]
Comment (broekhuisg) [reply frontend/containers/StorylineOverviewPage/StorylineOverviewPage.test.tsx]: I fixed it by adding data to this document and expecting that a string is returned

File: frontend/components/Storyline/StorylineOverview/StorylineOverview.tsx [hunk 4]
```diff
@@ -31,15 +60,22 @@ export default function StorylineOverview() {
...
+        xinformationTypes: [...new Set(obj.informationTypes.map(item => item.name))],
+      }))
+      .filter(project =>
+        selectedRoles.length > 0 ? selectedRoles.some(r => project.xroles.indexOf(r) >= 0) : project
+      )
+      .filter(project =>
+        selectedInformation.length > 0
+          ? selectedInformation.some(r => project.xinformationTypes.indexOf(r) >= 0)
+          : project
+      );
```
[the remark this answers is not in the document]
Comment (pascalalferink) [reply frontend/components/Storyline/StorylineOverview/StorylineOverview.tsx]: @antw this indeed looks a lot better, I changed the code

File: frontend/components/Storyline/StorylineOverview/StorylineOverviewCard.tsx (hunk 2 above)
Comment (pascalalferink) [reply frontend/components/Storyline/StorylineOverview/StorylineOverviewCard.tsx]: I agree and have improved it.

**Comment by pascalalferink** [comment]
All issues are solved,

1. The css is mostly converted into tailwind classes and for the exeptions a css module is created
2. For the open item (all storylines are included, also when they aren't a child of the current storyline overvie page) a new issue is created.

**Review by mattijsstam — APPROVED** [review]

**Review by broekhuisg — APPROVED** [review]
Approved

**Pull request merged**